parser_test.go

```diff
@@ -1812,6 +1812,20 @@ func (s *testParserSuite) TestDDL(c *C) {
 	s.RunTest(c, table)
 }
 
+func (s *testParserSuite) TestHintError(c *C) {
+	parser := New()
+	stmt, err := parser.Parse("select /*+ tidb_unknow(T1,t2) */ c1, c2 from t1, t2 where t1.c1 = t2.c1", "", "")
```

Comment:
I think we should return a warning to tell the user that the hint is unknown and takes no effect.

Reply:
I take it too, but it seems there are no elegance way to pass warn out of parser that in current situation....

so we should choose:

-  add a new error code then back code to 1064 (in mysql both error warn are 1064)
-  add a return value
-  using string.contains just like `try again later` :laughing: 

which way should we use? I think this category question maybe better disccuss before writing code :rocket: